Feat/landing page

Jidapa-Nakvisai opened the pull request:
Responsive home page (exclude map) & FAQ .

SiravitPhokeed commented on src/components/ui/accordion.tsx:
Personally I use [Motion for React](https://motion.dev/) for animations, it’s really good! You might want to check it out.

SiravitPhokeed reviewed (changes requested):
นัดเราไป vc ได้นะถ้าอยากได้ภาษาไทย

Some comments, but here’s the short version (and some more on things that happen everywhere):

- Please ask if you’re not sure about anything on the Figma file, It doesn't hurt to ask but it does hurt to assume.
- Please stick to the Figma file as much as possible. Don’t use colors not in the CI.
- A lot of the solutions seem very over-engineered, it might be better to pause and see if there’s a simpler solution before you start work on something!
- Be more consistent with your code. I see your `cn` class and I love the idea, but you seem to only use it in some places, for example.
- Ensure that every piece of code has a purpose. If you’re not sure why you added something, maybe you don’t need it.
- Prefer `rem` over `px`.
- Prefer Tailwind theme variables and custom utilities over arbitrary values.

I’m sorry it’s a little long. Wishing you the best!

Jidapa-Nakvisai commented on src/app/globals.css:
It's automatically got when installed shadcn ui component

Jidapa-Nakvisai commented on src/app/globals.css:
```diff
@@ -1,26 +1,275 @@
+  letter-spacing: -20px;
+}
+
+.lucide-chevron-down {
+  stroke: #a7271d;
 }
```
It's for the color of arrow of FAQ accordion

Jidapa-Nakvisai commented on package.json:
```diff
@@ -10,11 +11,19 @@
   },
   "dependencies": {
+    "@radix-ui/react-accordion": "^1.2.11",
+    "@svgr/webpack": "^8.1.0",
```
I was going to use it at first, but now it's no longer necessary. I will remove it

Jidapa-Nakvisai commented on src/components/ThemeColorUpdater.tsx:
```diff
@@ -0,0 +1,40 @@
+      }
+    };
+
+    requestAnimationFrame(updateThemeColor);
```
It's not necessary. Already removed.

Jidapa-Nakvisai commented on package.json:
```diff
@@ -10,11 +11,19 @@
+    "react-dom": "^19.0.0",
+    "styled-components": "^6.1.19",
+    "tailwind-merge": "^3.3.1",
+    "webpack": "^5.99.9"
```
I had some problems at first, but now it's ok. I will remove it

Jidapa-Nakvisai commented on package.json:
```diff
@@ -10,11 +11,19 @@
+    "clsx": "^2.1.1",
     "firebase": "^11.9.1",
     "firebase-admin": "^13.4.0",
+    "lucide-react": "^0.517.0",
```
Because I use shadcn, so it's installed automatically

SiravitPhokeed replied:
That doesn’t mean you can’t use Material Symbols. Shadcn makes getting started easier but you can still use your own stuff! That’s one of the selling points of Shadcn even iirc.

SiravitPhokeed commented on src/components/HomeWrapper.tsx:
```diff
@@ -0,0 +1,58 @@
+  return (
+    // Reuse component for home page and login page
+    <BackgroundWithNoise
+      classname={cn("bg-gradient-to-b from-yellow to-yellow-white", classname)}
```
I like this approach of factoring out the gradient to `className`!

SiravitPhokeed commented on src/app/globals.css:
```diff
@@ -1,26 +1,60 @@
+  --type-chinese-small: 2.25rem / 2.25rem var(--font-kart);
+  --type-chinese-cursive: 0.6rem / 6rem var(--font-kart);
+
+  --tracking-display-large: -0.25px;
+  --tracking-display-medium: 0;
+  --tracking-display-small: 0;
+  --tracking-headline-large: 0;
+  --tracking-headline-medium: 0;
+  --tracking-headline-small: 0;
+  --tracking-title-large: 0;
+  --tracking-title-medium: 0.15px;
+  --tracking-title-small: 0.1px;
+  --tracking-label-large: 0.1;
+  --tracking-label-medium: 0.5px;
+  --tracking-label-small: 0.5px;
+  --tracking-body-large: 0.5;
+  --tracking-body-medium: 0.25px;
+  --tracking-body-small: 0.4px;
+  --tracking-chinese-large: -3.5px;
+  --tracking-chinese-medium: -2px;
+  --tracking-chinese-small: -1px;
+  --tracking-chinese-cursive: -20px;
```
You can only omit `px` for `0`, everything else needs a unit! Please have another look.

SiravitPhokeed commented on package.json:
```diff
@@ -29,7 +36,8 @@
     "prettier": "^3.5.3",
-    "tailwindcss": "^4",
+    "tailwindcss": "^4.1.10",
+    "tw-animate-css": "^1.3.4",
```
Relying on `tw-animate-css` will be a problem down the line when you get to pages with more complex animations. Consider JavaScript animation libraries like [Motion](https://motion.dev) and [GSAP](https://gsap.com/). I personally find Motion very easy to use, much easier than CSS even!

Note that layout animations (called “Smart animate” in Figma) are **impossible to replicate with CSS**.

SiravitPhokeed commented on src/components/FaqCard.tsx:
```diff
@@ -0,0 +1,47 @@
+      >
+        <div className={`relative z-10 ${open ? "bg-yellow/20" : "bg-white"} `}>
+          <AccordionTrigger
+            className={`font-bold px-4 py-3 type-title-medium `}
```
Backticks aren’t necessary here.

SiravitPhokeed commented on src/app/page.tsx:
```diff
@@ -1,5 +1,51 @@
+          {MOCK_FAQ.map((item, index) => (
+            <div key={index} className="w-full  ">
+              {item.title.length > 0 && (
+                <div className="mb-4 bg-[url('/decorating/shapes/flag.svg')] bg-no-repeat w-42 h-10">
+                  <p className="type-title-medium font-bold text-left pl-7 py-2 text-white">
+                    {item.title}
+                  </p>
+                </div>
```
This might be better as a component since we’ll reuse it in the Manage Users page. 

Also you accidentally clipped the right edge.

![](https://github.com/user-attachments/assets/fd168e94-2363-4abf-9f11-da17f316170c)

SiravitPhokeed commented:
Oh, also, the commenter is usually the one that resolves the comments.

Jidapa-Nakvisai commented:
- removed unused import 
- removed IconProps file and changed src/app/types/FaqSectionProps.ts into faq.ts
- moved Accordion into app/page

- about Accordion  --> the required value attribute of AccordionItem is for track what item is open or close (same as key)
- about motion animation --> I'm not sure about layout animations. If I wrong, you can tell me!

Now, I think it's ready to be merged. If there are other issues, I'll fix them later!

SiravitPhokeed commented:
Here’s an infographic to explain using `layout` to create an accordion. If Radix UI doesn’t allow for this or you’re not proficient enough with it, it might be easier to make your own. All you need is 2 `<div>`s inside a `<div>` (if you don’t care about semantics!).

---

![layout transitions an element’s size. If an element changes from small to big, Motion will animate the in-between. layout="position" transitions an element’s position without changing its size. For an accordion, we want the top to stays fixed while we transition the accordion size. And how do we change the accordion size so Motion knows the animate the accordion opening? We start the content at height of 0 and change it to auto, and set the content to layout="position" so it doesn’t stretch. Notice how we used layout="position" so Motion doesn’t animate the height (which would be very expensive). To sum up, we start with this and end with this and Motion does the rest.](https://github.com/user-attachments/assets/13bbf36c-b6b5-438b-b520-f110c547a1bc)

---

The beauty of [Motion](https://motion.dev/docs/react-layout-animations) and other layout animation libraries is that it just looks at where an element is, compare it to where the element should be, and smoothly animates between the two without you having to figure out how that animation should work.
